src/services/ghost/classic-auto-complete/GhostContextProvider.ts

```diff
@@ -0,0 +1,89 @@
+import * as vscode from "vscode"
+import { ContextRetrievalService } from "../../continuedev/core/autocomplete/context/ContextRetrievalService"
+import { VsCodeIde } from "../../continuedev/core/vscode-test-harness/src/VSCodeIde"
+import { AutocompleteInput } from "../types"
+import { AutocompleteSnippetType } from "../../continuedev/core/autocomplete/snippets/types"
+import { HelperVars } from "../../continuedev/core/autocomplete/util/HelperVars"
+import { getAllSnippetsWithoutRace } from "../../continuedev/core/autocomplete/snippets/getAllSnippets"
+import { getDefinitionsFromLsp } from "../../continuedev/core/vscode-test-harness/src/autocomplete/lsp"
+import { DEFAULT_AUTOCOMPLETE_OPTS } from "../../continuedev/core/util/parameters"
+import { getSnippets } from "../../continuedev/core/autocomplete/templating/filtering"
+import { formatSnippets } from "../../continuedev/core/autocomplete/templating/formatting"
+
+function convertToContinuedevInput(autocompleteInput: AutocompleteInput) {
+	return {
+		...autocompleteInput,
+		recentlyVisitedRanges: autocompleteInput.recentlyVisitedRanges.map((range) => ({
+			...range,
+			type: AutocompleteSnippetType.Code,
+		})),
+	}
+}
+
+export class GhostContextProvider {
+	private contextService: ContextRetrievalService
+	private ide: VsCodeIde
+
+	constructor(context: vscode.ExtensionContext) {
+		this.ide = new VsCodeIde(context)
+		this.contextService = new ContextRetrievalService(this.ide)
+	}
+
+	/**
+	 * Get the IDE instance for use by tracking services
+	 */
+	public getIde(): VsCodeIde {
+		return this.ide
+	}
+
+	/**
+	 * Get context snippets for the current autocomplete request
+	 * Returns comment-based formatted context that can be added to prompts
+	 */
+	async getFormattedContext(autocompleteInput: AutocompleteInput, filepath: string): Promise<string> {
+		try {
+			// Convert filepath to URI if it's not already one
```

Comment:
when does this happen? Is vscode inconsistent in how it returns filepaths? that seems improbable